### PR TITLE
[Analytics Hub] Update networking to load data in parallel

### LINKED ISSUE #8318: [Analytics Hub] Separate Items Sold Network Request
For the Analytics Hub, we fire 3 network requests

- Current Order Stats
- Previous Order Stats
- Items Sold Stats

Currently, we are waiting on those 3 to finish before showing the non-redacted UI.

We could gain some perceived performance improvement if we separate the `Items Sold Stats` request as previous cards don't depend on them.

### PR DESCRIPTION
**Pull request by ealeksandrov**
Closes: https://github.com/woocommerce/woocommerce-ios/issues/8318.

## Description

This PR updates data loading flow to prevent waiting on all requests before updating any cards.

### How

- Update `TopPerformersView` to have its own viewmodel and loading state
- Update `AnalyticsProductCard` to accept 2 separate viewmodels
- Update data flow to trigger networking requests independently

## Testing

1. Build and run the app in debug/alpha mode.
2. On the dashboard view tap the "See more" button.
3. Scroll to the bottom of a list.
4. Confirm that top performers list has its own loading state and updates separately from the rest of stats.

## Video

> Revenue and orders cards are hidden to focus on products card

https://user-images.githubusercontent.com/3132438/206773285-0fde02e4-a885-4e85-9aff-6f5b8961042e.mp4

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/AnalyticsHubViewModel.swift [hunk 1]
```diff
@@ -103,26 +107,44 @@ final class AnalyticsHubViewModel: ObservableObject {
...
+        try await withThrowingTaskGroup(of: Void.self) { group in
+            group.addTask {
+                try await self.retrieveOrderStats(currentTimeRange: currentTimeRange, previousTimeRange: previousTimeRange)
+            }
+            group.addTask {
+                try await self.retrieveVisitorStats(currentTimeRange: currentTimeRange, previousTimeRange: previousTimeRange)
+            }
+            try await group.waitForAll()
+        }
+    }
```
Comment (Ecarrion) [inline WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/AnalyticsHubViewModel.swift:125]: Newbie question: Could we use a regular `async let` to keep requests more consistent, from reading at the code it is not clear to me why we need the task group, I'm sure we are missing something 😅

```swift
async let request1: () = retrieveOrderStats(currentTimeRange: currentTimeRange, previousTimeRange: previousTimeRange)
async let request2: () = retrieveVisitorStats(currentTimeRange: currentTimeRange, previousTimeRange: previousTimeRange)

_ = try await(request1, request2)
```

**Review by Ecarrion — APPROVED** [review]
Looks good to me! Just a newbie question about the task group.

File: WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/AnalyticsHubViewModel.swift (hunk 1 above)
Comment (ealeksandrov) [reply WooCommerce/Classes/ViewRelated/Dashboard/Analytics Hub/AnalyticsHubViewModel.swift:125]: Good question! I agree that simplest case of "start parallel requests and don't manage them at all" looks pretty complicated in async/await.

Your suggestion was one of my attempts, but in my testing it stopped all parallel requests when only one of them errored. I'll add a couple of unit tests for edge cases I've noticed.
